modules/geo/src/test/java/org/opensearch/geo/search/aggregations/bucket/composite/GeoTileGridAggregationCompositeAggregatorTests.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geo.search.aggregations.bucket.composite;
+
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.LatLonPoint;
+import org.apache.lucene.document.SortedNumericDocValuesField;
+import org.apache.lucene.search.DocValuesFieldExistsQuery;
+import org.apache.lucene.search.MatchAllDocsQuery;
+import org.junit.Before;
+import org.opensearch.common.geo.GeoPoint;
+import org.opensearch.geo.GeoModulePlugin;
+import org.opensearch.geo.search.aggregations.bucket.geogrid.GeoTileGridAggregationBuilder;
+import org.opensearch.geo.search.aggregations.bucket.geogrid.GeoTileGridAggregator;
+import org.opensearch.index.mapper.GeoPointFieldMapper;
+import org.opensearch.plugins.SearchPlugin;
+import org.opensearch.search.aggregations.bucket.composite.CompositeAggregationBuilder;
+import org.opensearch.search.aggregations.bucket.GeoTileUtils;
+import org.opensearch.search.aggregations.composite.BaseCompositeAggregatorTestCase;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Testing the {@link GeoTileGridAggregator} as part of CompositeAggregation.
+ */
+public class GeoTileGridAggregationCompositeAggregatorTests extends BaseCompositeAggregatorTestCase {
+
+    protected List<SearchPlugin> getSearchPlugins() {
+        return Collections.singletonList(new GeoModulePlugin());
+    }
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        FIELD_TYPES.add(new GeoPointFieldMapper.GeoPointFieldType("geo_point"));
+    }
+
+    public void testUnmappedFieldWithGeopoint() throws Exception {
+        final List<Map<String, List<Object>>> dataset = new ArrayList<>();
+        final String mappedFieldName = "geo_point";
+        dataset.addAll(
+            Arrays.asList(
+                createDocument(mappedFieldName, new GeoPoint(48.934059, 41.610741)),
```

Comment:
can this be generated randomly?

Reply:
This test case was already like this. But for this specific one I will make the changes. For the next test case in the same file it won't be possible,